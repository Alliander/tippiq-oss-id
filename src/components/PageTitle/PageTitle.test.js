import React from 'react';
import renderer from 'react-test-renderer';
import Element from './PageTitle';

test('with plain text title', () => {
  const component = renderer.create(
    <Element>Title</Element>
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});

test('with nested elements in title', () => {
  const component = renderer.create(
    <Element>Title <span>with</span> nesting</Element>
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});

test('with id', () => {
  const component = renderer.create(
    <Element id="test-id">Title</Element>
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});

test('with null id', () => {
  const component = renderer.create(
    <Element id={null}>Title</Element>
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
