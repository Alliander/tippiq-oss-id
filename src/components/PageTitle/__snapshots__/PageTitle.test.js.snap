// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with id 1`] = `
<h1
  className={undefined}
  id="test-id"
>
  Title
</h1>
`;

exports[`with nested elements in title 1`] = `
<h1
  className={undefined}
>
  Title 
  <span>
    with
  </span>
   nesting
</h1>
`;

exports[`with null id 1`] = `
<h1
  className={undefined}
>
  Title
</h1>
`;

exports[`with plain text title 1`] = `
<h1
  className={undefined}
>
  Title
</h1>
`;
